pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -0,0 +1,420 @@
+from pyscript import display
+
+# from pyscript import when as _when
+
+alert = js.alert
+
+
+class BaseElement:
+    def __init__(self, js_element):
+        self._element = js_element
+        self._parent = None
+        self.style = StyleProxy(self)
+
+    def __eq__(self, obj):
+        """Check if the element is the same as the other element by comparing
+        the underlying JS element"""
+        return isinstance(obj, BaseElement) and obj._element == self._element
+
+    @property
+    def parent(self):
+        if self._parent:
+            return self._parent
+
+        if self._element.parentElement:
+            self._parent = self.__class__(self._element.parentElement)
+
+        return self._parent
+
+    @property
+    def __class(self):
+        return self.__class__ if self.__class__ != PyDom else Element
+
+    def create(self, type_, is_child=True, classes=None, html=None, label=None):
+        js_el = js.document.createElement(type_)
+        element = self.__class(js_el)
+
+        if classes:
+            for class_ in classes:
+                element.add_class(class_)
+
+        if html is not None:
+            element.html = html
+
+        if label is not None:
+            element.label = label
+
+        if is_child:
+            self.append(element)
+
+        return element
+
+    # -------- Boilerplate Proxy for the Element API -------- #
+    @property
+    def appendChild(self):
+        return self._element.appendChild
+
+
+class Element(BaseElement):
+    def append(self, child):
+        # TODO: this is Pyodide specific for now!!!!!!
+        # if we get passed a JSProxy Element directly we just map it to the
+        # higher level Python element
+        if isinstance(child, JsProxy):
+            return self.append(self.from_js(child))
+
+        elif isinstance(child, Element):
+            self.appendChild(child._element)
+
+            return child
+
+    def from_js(self, js_element):
+        return self.__class__(js.tagName, parent=self)
+
+    def query(self, selector):
+        """The querySelector() method of the Element interface returns the first
+        element that is a descendant of the element on which it is invoked that
+        matches the specified group of selectors.
+        """
+        return self.__class__(self._element.querySelector(selector))
+
+    def query_all(self, selector):
+        """The querySelectorAll() method of the Element interface returns a static
+        (not live) NodeList representing a list of the document's elements that
+        match the specified group of selectors.
+        """
+        for element in self._element.querySelectorAll(selector):
+            yield self.__class__(element)
+
+    # -------- Boilerplate Proxy for the Element API -------- #
+    @property
+    def html(self):
+        return self._element.innerHTML
+
+    @html.setter
+    def html(self, value):
+        self._element.innerHTML = value
+
+    @property
+    def content(self):
```

Comment:
for `template` element `content` returns a *DocumentFragment*

```js
const tpl = document.createElement('template');
tpl.innerHTML = '<div></div>';
tpl.content;
// #document-fragment
//  <div></div>
```

this breaks with standard template elements but it might be confusing out of documentation searches on Google and friends

Reply:
We'll be using `template` elements in the invent framework.

Reply:
Yes. This is a temporary exclusion to cover the majority of cases and understand what's the right thing to do for template before we actually support it. 👍

Reply:
you already support it (breaking it somehow) by accepting any DOM element and a `<template>` is a DOM element, with an ID and everything else ... templates need extra care in here and be addressed as `reference._js.content` instead to not mislead, I hope that will be documented somehow.